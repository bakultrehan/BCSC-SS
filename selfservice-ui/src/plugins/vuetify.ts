import Vue from 'vue';
import Vuetify from 'vuetify/lib';

Vue.use(Vuetify);

export default new Vuetify({
  icons: {
    iconfont: 'mdi',
  },
  theme: {
    themes: {
      light: {
        error: '#D8292F',
        success: '#2E8540',
        warning: '#FFC107',
      },
    },
    options: {
      cspNonce: 'cLv1G2y8KtMdR',
    },
  },
});
