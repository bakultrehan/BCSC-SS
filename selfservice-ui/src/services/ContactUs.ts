import axios from '@/lib/axios';
import { ContactDetailsModel } from '@/models/ContactDetailsModel';
import { CONTACTUS_URL } from '@/config/api-endpoints';
export class ContactUs {
  public static async addContactUs(contactDetails: ContactDetailsModel) {
    return await axios.post(CONTACTUS_URL, contactDetails);
  }
}
