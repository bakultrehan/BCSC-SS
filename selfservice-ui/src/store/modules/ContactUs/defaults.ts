const contactDetailsSample = () => {
  return {
    firstName: '',
    lastName: '',
    email: '',
    description: '',
  };
};

export { contactDetailsSample };
