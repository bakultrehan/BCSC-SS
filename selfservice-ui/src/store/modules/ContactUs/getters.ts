import { GetterTree } from 'vuex';
import { TestAccountState } from './types';
import { RootState } from '../../types';

/**
 * Getters Shared
 */
export const getters: GetterTree<TestAccountState, RootState> = {
  /**
   * isLoading  status
   * @param {*} state
   * @returns {boolean} - loading status
   */
  isLoading(state) {
    return state.isLoading;
  },
  getChangeStatus(state) {
    return {
      successStatus: state.successStatus,
      errorStatus: state.errorStatus,
    };
  },
};
