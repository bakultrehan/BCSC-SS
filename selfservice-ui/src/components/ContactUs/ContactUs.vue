/** 
* 
ContactUs component 
*/

<template>
  <v-card class="mx-auto outer-card">
    <v-form ref="form" v-model="valid">
      <h1 class="bc-page-title-h1">{{ $t('contactUs.pagetitle') }}</h1>
      <div class="flex-grow-1"></div>

      <v-container>
        <v-row class="mx-4">
          <v-col cols="12" flat>
            <v-card flat>
              <div class="text-left" v-html="$t('contactUs.pageinfo')"></div>
            </v-card>
          </v-col>
          <v-col class="col-12" v-if="errorStatus || successStatus">
            <Alert
              type="error"
              v-if="errorStatus"
              class="alert-top text-left"
            >{{ $t('contactUs.errorMessage') }}</Alert>
            <Alert
              type="success"
              class="alert-top text-left"
              v-if="successStatus"
            >{{ $t('contactUs.successMessage') }}</Alert>
          </v-col>
          <v-col cols="8">
            <Input
              v-model="contactDetails.firstName"
              :label="$t('contactUs.firstName')"
              type="text"
              :rules="[rules.required,  rules.maxLength(500)]"
              data-test-id="input-firstname"
            />

            <Input
              v-model="contactDetails.lastName"
              :label="$t('contactUs.lastName')"
              type="text"
              :rules="[rules.required, rules.maxLength(500)]"
              data-test-id="input-lastname"
            />
            <Input
              v-model="contactDetails.email"
              :label="$t('contactUs.emailAddress')"
              type="text"
              :rules="[rules.required, rules.email, rules.maxLength(500)]"
              data-test-id="input-email"
            />
          </v-col>
          <v-col cols="12" flat>
            <v-card flat>
              <TextArea
                v-model="contactDetails.description"
                :label="$t('contactUs.description')"
                type="text"
                outlined
                rows="10"
                name="text-contact-us-description"
                id="text-contact-us-description"
                data-test-id="text-contact-us-description"
                :rules="[rules.required]"
              />
            </v-card>
          </v-col>
        </v-row>

        <v-col cols="12">
          <v-card flat>
            <v-divider></v-divider>
            <v-card-actions>
              <v-spacer></v-spacer>

              <Button
                :disabled="!valid"
                class="white--text btn-contact-us ml-6"
                depressed
                @click="submitContactMessage"
                name="btn-contact-us"
                data-test-id="btn-contact-us"
              >{{ $t('contactUs.btnSend') }}</Button>
            </v-card-actions>
          </v-card>
        </v-col>
      </v-container>
    </v-form>
  </v-card>
</template>
<script lang="ts">
import { Component, Vue, Watch, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import validationRules from '@/config/validationRules';

import Button from '@/Atomic/Button/Button.vue';
import TextArea from '@/Atomic/TextArea/TextArea.vue';
import Alert from '@/Atomic/Alert/Alert.vue';
import Input from '@/Atomic/Input/Input.vue';

import { contactDetailsSample } from '@/store/modules/ContactUs/defaults';

const ContactUsModule = namespace('ContactUsModule');

@Component({
  components: {
    Button,
    TextArea,
    Alert,
    Input
  }
})
export default class ContactUs extends Vue {
  @Prop({ default: 0 })
  public id!: number;

  @ContactUsModule.Action('addContactUs')
  public addContactUs!: any;
  @ContactUsModule.Getter('isLoading')
  public isLoading!: boolean;
  @ContactUsModule.Getter('getChangeStatus')
  public getChangeStatus!: boolean;

  private rules = validationRules;

  private contactDetails: any;
  private valid: boolean = false;
  private successStatus: boolean = false;
  private errorStatus: boolean = false;

  public constructor() {
    super();
    this.contactDetails = contactDetailsSample();
  }

  @Watch('getChangeStatus')
  private ongetChangeStatusChanged(val: any) {
    const { successStatus, errorStatus } = val;
    this.successStatus = successStatus;
    this.errorStatus = errorStatus;
    if (successStatus) {
      (this.$refs.form as Vue & {
        reset: () => any;
      }).reset();
      this.$vuetify.goTo(0, {
        duration: 1000,
        easing: 'easeInOutCubic'
      });
    }
  }

  private submitContactMessage() {
    this.addContactUs({
      contactDetails: this.contactDetails
    });
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
</style>
