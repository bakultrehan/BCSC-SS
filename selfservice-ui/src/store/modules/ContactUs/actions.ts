import { ActionTree } from 'vuex';
import { TestAccountState } from './types';
import { RootState } from '../../types';
import { ContactUs } from '@/services/ContactUs';

/**
 * contact us Actions
 *
 */
export const actions: ActionTree<TestAccountState, RootState> = {
  /**
   * @param  {} {commit}
   * @param  {} data contact us data
   */
  async addContactUs({ commit, dispatch }, data: any) {
    commit('SET_LOADING', true);
    dispatch('clearStatus');

    try {
      const { contactDetails } = data;
      await ContactUs.addContactUs({ ...contactDetails });
      commit('SET_SUCESS', true);
      commit('SET_ERROR', false);
      commit('SET_LOADING', false);
    } catch {
      commit('SET_ERROR', true);
      commit('SET_SUCESS', false);
      commit('SET_LOADING', false);
    }
  },
  /**
   * clear message
   * @param {*} { commit }
   */
  async clearStatus({ commit }) {
    commit('SET_SUCESS', false);
    commit('SET_ERROR', false);
    commit('SET_LOADING', false);
  },
};
