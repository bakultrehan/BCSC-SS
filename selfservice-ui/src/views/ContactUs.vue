<template>
  <v-card flat>
    <ContactUs />
  </v-card>
</template>

<script>
// @ is an alias to /src
import { Component, Vue } from 'vue-property-decorator';
import ContactUs from '@/components/ContactUs/ContactUs.vue';

@Component({
  components: {
    ContactUs
  }
})
// Home goes here
export default class ContactUsPage extends Vue {}
</script>
