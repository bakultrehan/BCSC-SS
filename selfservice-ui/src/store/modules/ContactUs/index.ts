import { Module } from 'vuex';
import { getters } from './getters';
import { actions } from './actions';
import { mutations } from './mutations';
import { TestAccountState } from './types';
import { RootState } from '../../types';
import { contactDetailsSample } from './defaults';

/**
 * Sharedmodule state
 */
export const state: TestAccountState = {
  contactDetails: contactDetailsSample(),
  isLoading: false,
  successStatus: false,
  errorStatus: false,
};

const namespaced: boolean = true;

const SharedModule: Module<TestAccountState, RootState> = {
  namespaced,
  state,
  getters,
  actions,
  mutations,
};

export default SharedModule;
