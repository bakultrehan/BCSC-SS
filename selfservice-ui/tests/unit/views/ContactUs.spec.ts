import { shallowMount, createLocalVue } from '@vue/test-utils';
import ContactUs from '@/views/ContactUs.vue';
import vuetify from 'vuetify';

describe('ContactUs.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(vuetify);

    wrapper = shallowMount(ContactUs, {
      localVue,
    });
  });

  it('renders props when passed', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
