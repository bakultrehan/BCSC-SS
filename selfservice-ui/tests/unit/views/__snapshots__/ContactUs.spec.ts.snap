// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUs.vue renders props when passed 1`] = `
<v-card-stub
  flat="true"
  loaderheight="4"
  tag="div"
>
  <contactus-stub
    id="0"
  />
</v-card-stub>
`;
